src/renderer/store/selectors/channel.ts

```diff
@@ -161,46 +176,40 @@ export const shareableUri = createSelector(channel, c => c.shareableUri)
 export const channelId = createSelector(channel, ch => ch.id)
 
 export const inputLocked = createSelector(
-  identitySelectors.balance('zec'),
-  // identitySelectors.lockedBalance('zec'),
-  // users.users,
-  // identitySelectors.signerPubKey,
   channelId,
-  contacts,
+  directMessagesSelectors.users,
+  waggleSelectors.isConnected,
+  isPublicChannel,
   (
-    available,
-    // locked,
-    // users,
-    // signerPubKey,
     channelId,
-    contacts
+    waggleContacts,
+    waggle,
+    publicChannel
   ) => {
-    const contactsData = Object.values(contacts)
+    const contactsData = Object.values(waggleContacts)
     const currentContactArray = contactsData.filter(item => {
-      return item.key === channelId && (item.connected === (false || true))
+      return item.publicKey === channelId
     })
 
-    if (currentContactArray[0]) {
-      if (currentContactArray[0].connected) {
+    if (waggle) {
```

Comment:
We'd have less nesting if we first checked if (!waggle). 
You could merge publicChannel and currentContactArray[0] conditional.